src/VisualStudio/CSharp/Impl/Interactive/CSharpInteractiveWindowCommandCompletionProvider.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion.Providers;
+using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.InteractiveWindow;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
+{
+    /// <summary>
+    /// Provides completion items for Interactive Window commands (such as #help, #cls, etc.) at the start of a C# language buffer.
+    /// </summary>
+    [ExportCompletionProviderMef1("ReplCommandCompletionProvider", LanguageNames.CSharp)]
+    [TextViewRole(PredefinedInteractiveTextViewRoles.InteractiveTextViewRole)]
+    [Order(Before = PredefinedCompletionProviderNames.Keyword)]
+    internal sealed class CSharpInteractiveWindowCommandCompletionProvider : AbstractInteractiveWindowCommandCompletionProvider
```

Comment:
IsPreProcessorKeywordContext is C# specific. Since both the commands and C# preproc directives have the same syntax there is an interplay between them.

Reply:
Maybe it can be changes/improved, but I'd rather leave refactoring of the commands for later.